funding/manager.go

```diff
@@ -1022,8 +1025,14 @@ func (m *Manager) OfferSidecar(ctx context.Context, capacity,
 		return nil, fmt.Errorf("error signing offer: %v", err)
 	}
 
-	// Let's now store and return the ticket with the signed offer.
-	err = m.cfg.DB.AddSidecar(ticket)
+	// Let's now store and return the ticket with the signed offer. If a
+	// bid was specified, then we'll commit that as well so we can submit
+	// it to the auctioneer later once we communicate with the recipient.
+	if bid == nil {
+		err = m.cfg.DB.AddSidecar(ticket)
+	} else {
+		err = m.cfg.DB.AddSidecarWithBid(ticket, bid)
```

Comment:
hm, could we unify this to _always_ require a bid with the sidecar?

Reply:
Yeah I tried that initially, but the trouble I ran into was a circular dependnacy: we can't add a bid to the main interface w/o introducing one. So it would need some bigger overhaul to pull off...

Reply:
Yeah, had some similar troubles too, that's why the order nonce is `[32]byte` and not `order.Nonce`. The closest we can come is to add a new `Bid` struct to the sidecar that uses primitive types only and then in the `order` package have a `NewBidFromTicket` function or something.

Reply:
Hmm, is that really a better option though? As is we don't have a routine that serializes a bid into a _single_ byte stream. Instead, due to historical reasons, we store them strewn amongst a few sub buckets, etc. Therefore it may be the case that the way we store the bid serialized within a sidecar starts to get out of sync with the way we store it on disk. With what's implemented right now, we get the benefit of re-using the exact set of storage routines both in this new sub-bucket and elsewhere.